### PR TITLE
remove 'unless' condition to ensure group add/rm works

### PR DESCRIPTION
**Pull request by noelmcloughlin**
Fix this unexpected issue on Fedora27 when Active Directory is used
```
----------
          ID: openstack-devstack cleandown
    Function: group.absent
        Name: stack
      Result: False
     Comment: Failed to remove group stack
     Started: 19:36:30.167747
    Duration: 50.518 ms
     Changes:   


          ID: openstack-devstack ensure user and group exist
    Function: group.present
        Name: stack
      Result: True
     Comment: unless execution succeeded
     Started: 19:36:30.245230
    Duration: 31.529 ms
     Changes:   
----------
          ID: openstack-devstack ensure user and group exist
    Function: user.present
        Name: stack
      Result: False
     Comment: The following group(s) are not present: stack
     Started: 19:36:30.283611
    Duration: 242.782 ms
     Changes:

```

### CODE REVIEW

File: devstack/clean.sls [hunk 1]
```diff
@@ -32,20 +32,17 @@ openstack-devstack cleandown:
   user.absent:
     - name: {{ devstack.local.username }}
     - purge: True
-    - onlyif: getent passwd {{ devstack.local.username }}
   cmd.run:
```
Comment (javierbertoli) [inline devstack/clean.sls]: Isn't this `cmd.run` redundant because of the `user.absent` `purge:true`?

File: devstack/init.sls [hunk 2]
```diff
@@ -12,7 +12,6 @@ openstack-devstack ensure package dependencies:
 openstack-devstack ensure user and group exist:
```
Comment (javierbertoli) [inline devstack/init.sls]: Perhaps user management might be moved to its own state file `users.sls`, and extended/included here and in `clean.sls`, in order to keep follow DRY?

**Review by javierbertoli — COMMENTED** [review]
@noelmcloughlin, just a couple of questions/comments.

File: devstack/clean.sls (hunk 1 above)
Comment (noelmcloughlin) [reply devstack/clean.sls]: Yes it's always redundant unless `user.absent `state fails causing cmd.run (onfail) to be triggered.  I recall adding cmd.run due to some unexpected results from user.absent (possibly related to fact this server uses Active Directory, or an environment problem).

File: devstack/init.sls (hunk 2 above)
Comment (noelmcloughlin) [reply devstack/init.sls]: That's a reasonable suggestion - done!!

File: devstack/clean.sls (hunk 1 above)
Comment (javierbertoli) [reply devstack/clean.sls]: Ah, gotcha. I thought it was, perhaps, due to the fact that the user was still in use (where a `force in user.absent` would help?).

Comment (noelmcloughlin) [reply devstack/clean.sls]: Ahh to have the time to figure out cause of every problem ;-)   Adding `force: true` is no harm, added!! thanks for the review comments.

**Comment by javierbertoli** [comment]
Great work, @noelmcloughlin

**Pull request merged**